starport/cmd/network_chain_publish.go

```diff
@@ -34,6 +35,7 @@ func NewNetworkChainPublish() *cobra.Command {
 	c.Flags().String(flagTag, "", "Git tag to use for the repo")
 	c.Flags().String(flagHash, "", "Git hash to use for the repo")
 	c.Flags().String(flagGenesis, "", "URL to a custom Genesis")
+	c.Flags().Uint64(flagCampaign, 0, "Campaign ID to use for this network")
```

Comment:
A campaign ID can be 0

Reply:
Why, can we avoid that for the sake of simplicity?

Reply:
It's a `list` type so it uses an incremental ID. Can't we use `-1` for the undefined flag?

Reply:
I like the idea of avoiding the id zero. This is can help us to avoid some checks

Reply:
I think it's not very hard to skip `id=0` for campaign on `spn` although I find it a bit unclean in a implementation detail point of view since every other list start with 0 so far

The main concern for me is that it would make the implementation of Starport CLI more ambiguous. Using `""` as a undefined value for a string variable is I may think a widely used convention, `""` explicitly tells that there is no string. While `0` it's more ambiguous, `0` is a `uint64` like any other integer, this is not explicit that this represent the undefined value for integer while `-1` or `nil` do. There shouldn't be technical concern but I'm arait that we take shortcut that make the implementation more ambiguous

Reply:
In Go `""` means zero value (not undefined) for strings and `0` is the zero value for numbers except floats. I think, it would be much proper if we avoid using zero values in the SPN implementation since it has a special meaning in Go and zero values usually interpreted by the Go packages in a special way. 

Whether changing from 0 to 1 in scaffolding (list) I think is not related to this topic but we can also track it by another issue.

Reply:
I'm not talking about Go 0 value, it's about variables that are implicitly considered as non-initialized.
Independently of Go, having id starting at 0 and that can get assigned this value is a widely used convention and having methods `func foo(..., campaignID uint64)` where `0` means no campaign is, in my opinion, an internal rule that would make the codebase more complicated to understand

Reply:
I'm no sure if there is such thing like implicitly considered non initialized value in Go, it is basically called zero values. And zero value for an integer is 0.

I don't think we need to get lost in transition. It seems to me that it's not essential to use 0 as an id in the SPN or anything that we scaffold. In-fact it makes things easier to avoid using zero values for entries. Using them actually creates confusion because there is no a null in Go, zero values usually used in place of null for comparing if a variable has a value.